util/src/builder/command.rs

```diff
@@ -0,0 +1,759 @@
+            description,
+            name,
+            required: false,
+        })
+    }
+
+    /// Consume the builder, returning the built command option.
+    #[allow(clippy::missing_const_for_fn)]
+    #[must_use = "should be used in a command builder"]
+    pub fn build(self) -> CommandOption {
+        CommandOption::Boolean(self.0)
+    }
+
+    /// Set whether this option is required.
+    ///
+    /// Defaults to false.
+    pub const fn required(mut self, required: bool) -> Self {
+        self.0.required = required;
+
+        self
+    }
+}
+
+impl From<BooleanBuilder> for CommandOption {
+    fn from(builder: BooleanBuilder) -> CommandOption {
+        builder.build()
+    }
+}
+
+/// Create a channel option with a builder.
+#[derive(Clone, Debug)]
+#[must_use = "should be used in a command builder"]
+pub struct ChannelBuilder(BaseCommandOptionData);
+
+impl ChannelBuilder {
+    /// Create a new default [`ChannelBuilder`].
+    #[must_use = "builders have no effect if unused"]
+    pub const fn new(name: String, description: String) -> Self {
+        Self(BaseCommandOptionData {
+            description,
+            name,
+            required: false,
+        })
+    }
+
+    /// Consume the builder, returning the built command option.
+    #[allow(clippy::missing_const_for_fn)]
+    #[must_use = "should be used in a command builder"]
+    pub fn build(self) -> CommandOption {
+        CommandOption::Channel(self.0)
+    }
+
+    /// Set whether this option is required.
+    ///
+    /// Defaults to false.
+    pub const fn required(mut self, required: bool) -> Self {
+        self.0.required = required;
+
+        self
+    }
+}
+
+impl From<ChannelBuilder> for CommandOption {
+    fn from(builder: ChannelBuilder) -> CommandOption {
+        builder.build()
+    }
+}
+/// Create a integer option with a builder.
+#[derive(Clone, Debug)]
+#[must_use = "should be used in a command builder"]
+pub struct IntegerBuilder(ChoiceCommandOptionData);
+
+impl IntegerBuilder {
+    /// Create a new default [`IntegerBuilder`].
+    #[must_use = "builders have no effect if unused"]
+    pub const fn new(name: String, description: String) -> Self {
+        Self(ChoiceCommandOptionData {
+            choices: Vec::new(),
+            description,
+            name,
+            required: false,
+        })
+    }
+
+    /// Consume the builder, returning the built command option.
+    #[allow(clippy::missing_const_for_fn)]
+    #[must_use = "should be used in a command builder"]
+    pub fn build(self) -> CommandOption {
+        CommandOption::Integer(self.0)
+    }
+
+    /// Add a list of choices to the command.
+    ///
+    /// Defaults to no choices.
+    pub fn choices(mut self, choices: impl IntoIterator<Item = (String, i64)>) -> Self {
+        self.0.choices = choices
+            .into_iter()
+            .map(|(name, value)| CommandOptionChoice::Int { name, value })
+            .collect();
+
+        self
+    }
```

Comment:
The documentation says "add a list of choices" but the command **overwrites** the choices. We should either update the docs or append the choices:
```rust
for choice in choices.into_iter().map(|name, value)| CommandOptionChoice::Int { name, value }) {
    self.0.choices.push(choice);
}
```

Reply:
I'll update the documentation